certbot/tests/client_test.py

```diff
@@ -67,8 +67,7 @@ def test_email_retry(self, _rep, mock_get_email):
         mx_err = messages.Error(detail=msg, typ="urn:acme:error:invalidEmail")
         with mock.patch("certbot.client.acme_client.Client") as mock_client:
             mock_client().register.side_effect = [mx_err, mock.MagicMock()]
-            self._call()
-            self.assertEqual(mock_get_email.call_count, 1)
+            self.assertRaises(errors.MissingCommandlineFlag, self._call)
```

Comment:
Hmmm. I think you want to test this error-raising property (in the case where you're in non-interactive mode), but keep the previous test in place (in the case where you're not in non-interactive mode).

If I had to guess, the test probably failed because we were defaulting to being in `non-interactive` mode due to `mock`ing things?